packages/flutter_tools/test/commands.shard/hermetic/project_validator_test.dart

```diff
@@ -0,0 +1,92 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// @dart = 2.8
+
+import 'package:args/command_runner.dart';
+import 'package:file/memory.dart';
+import 'package:flutter_tools/src/base/file_system.dart';
+import 'package:flutter_tools/src/base/logger.dart';
+import 'package:flutter_tools/src/commands/validate_project.dart';
+import 'package:flutter_tools/src/project.dart';
+import 'package:flutter_tools/src/project_validator.dart';
+import 'package:flutter_tools/src/project_validator_result.dart';
+
+import '../../src/context.dart';
+import '../../src/test_flutter_command_runner.dart';
+
+class ProjectValidatorDummy extends ProjectValidator {
+  @override
+  Future<List<ProjectValidatorResult>> start(FlutterProject project) async{
+    return <ProjectValidatorResult>[
+      const ProjectValidatorResult(name: 'pass', value: 'value', status: StatusProjectValidator.success),
+      const ProjectValidatorResult(name: 'fail', value: 'my error', status: StatusProjectValidator.error),
+      const ProjectValidatorResult(name: 'pass two', value: 'pass', warning: 'my warning', status: StatusProjectValidator.warning),
+    ];
+  }
+
+  @override
+  bool supportsProject(FlutterProject project) {
+    return true;
+  }
+}
+
+class ProjectValidatorCrash extends ProjectValidator {
+  @override
+  Future<List<ProjectValidatorResult>> start(FlutterProject project) async{
+    throw Exception('my exception');
+  }
+
+  @override
+  bool supportsProject(FlutterProject project) {
+    return true;
+  }
+}
+
+void main() {
+  final BufferLogger loggerTest = BufferLogger.test();
+  FileSystem fileSystem;
+
+  group('analyze project command', () {
+    setUpAll(() {
```

Comment:
Don't re-use the same filesystem between tests

```suggestion
    setUp(() {
```